src/m365/outlook/commands/message/message-list.ts

```diff
@@ -67,19 +137,40 @@ class OutlookMessageListCommand extends GraphCommand {
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     try {
-      const folderId = await this.getFolderId(args);
+      if (!args.options.userId && !args.options.userName && accessToken.isAppOnlyAccessToken(auth.connection.accessTokens[this.resource].accessToken)) {
+        throw 'You must specify either the userId or userName option when using an app-only access token';
+      }
+
+      const userUrl = args.options.userId || args.options.userName ? `users/${args.options.userId || args.options.userName}` : 'me';
+
+      const folderId = await this.getFolderId(userUrl, args);
+      const folderUrl: string = folderId ? `/mailFolders/${folderId}/messages` : '/messages';
+      let requestUrl = `${this.resource}/v1.0/${userUrl}${folderUrl}`;
 
-      const url: string = folderId ? `me/mailFolders/${folderId}/messages` : 'me/messages';
-      const messages = await odata.getAllItems<Message>(`${this.resource}/v1.0/${url}?$top=50`);
+      if (args.options.startTime || args.options.endTime) {
+        const filters = [];
+
+        if (args.options.startTime) {
+          filters.push(`receivedDateTime ge ${formatting.encodeQueryParameter(args.options.startTime)}`);
```

Comment:
I'm not sure if it's needed to encode the date since it doesn't really contain special characters.